src/main/java/org/broadinstitute/dsm/util/externalShipper/GBFRequestUtil.java

```diff
@@ -338,8 +327,59 @@ public void orderConfirmation(long startDate, long endDate) throws Exception {
     }
 
     public void orderCancellation(ArrayList<KitRequest> kitRequests) throws Exception {
+        throw new NotImplementedException("Not implemented yet");
+    }
+
+    public void updateOrderStatusForPendingKitRequests(int instanceId) {
+        updateOrderStatusForPendingKitRequests(instanceId, SQL_SELECT_EXTERNAL_KIT_NOT_DONE);
+    }
+
+    /**
+     * Streams through the list of pending requests from the
+     * database and updates the status based on the latest
+     * details from GBF
+     */
+    private void updateOrderStatusForPendingKitRequests(int instanceId, String query) {
+        final AtomicInteger numOrdersProcessed = new AtomicInteger();
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(query)) {
+                stmt.setInt(1, instanceId);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    while (rs.next()) {
+                        String ddpParticipantId = rs.getString(DBConstants.DDP_PARTICIPANT_ID);
+                        if (StringUtils.isNotBlank(ddpParticipantId)) {
+                            KitRequest kitRequest = new KitRequest(rs.getString(DBConstants.DSM_KIT_REQUEST_ID), ddpParticipantId,
+                                    null, null, rs.getString(DBConstants.EXTERNAL_ORDER_NUMBER), null,
+                                    rs.getString(DBConstants.EXTERNAL_ORDER_STATUS),
+                                    rs.getString("subkits." + DBConstants.EXTERNAL_KIT_NAME),
+                                    rs.getLong(DBConstants.EXTERNAL_ORDER_DATE));
+                            numOrdersProcessed.incrementAndGet();
+                            try {
+                                orderStatus(conn, kitRequest);
```

Comment:
Wrapping this in a try/catch so we can keep going even if we run into a problem for a single `kitRequest`.